crawler/helpers/general_helpers.py

```diff
@@ -253,6 +254,31 @@ def map_mongo_sample_to_mysql(doc: SampleDoc, copy_date: bool = False) -> Dict[s
     return value
 
 
+def set_is_current_on_mysql_samples(samples: Iterable[Dict[str, str]]) -> List[Dict[str, str]]:
+    """Creates a copy of the samples passed in, adding is_current values to each sample.
+    is_current will be True for all samples unless there is a repeated RNA ID, in which case
+    only the last one is set to True.
+
+    Arguments:
+        samples: Iterable[Dict[str, str]] -- An iterable containing dictionaries of sample data.
+
+    Returns:
+        A list containing new copies of the samples in the same order with is_current populated for each.
+    """
+    reversed_samples: List[Dict[str, Any]] = []
+    existing_rna_ids = set([""])
+    for sample in reversed(list(samples)):
+        try:
+            rna_id = sample[MLWH_RNA_ID]
+            reversed_samples.append({**sample, MLWH_IS_CURRENT: rna_id not in existing_rna_ids})
+            existing_rna_ids.add(rna_id)
+        except KeyError:
+            # If there is no RNA ID (shouldn't happen) set is_current to False
+            reversed_samples.append({**sample, MLWH_IS_CURRENT: False})
+
+    return list(reversed(reversed_samples))
```

Comment:
Why do you use reversed?

Reply:
Because we can assume all samples should have `is_current = True` unless there is a duplicate RNA ID later in the file.  We don't know what's later in the file, so it's easier to work from the bottom to the top.  Keep setting `is_current = True` on each entry unless we get to an RNA ID we already processed in this file, in which case set `is_current = False`.  Once we've processed the file from bottom to top, flip it back over again before returning.

Reply:
Ah, now I see; I would add it as a comment then before the loop.